### PR TITLE
Implement exercise markdown

### LINKED ISSUE #729: markdown: implement exercise
There are exercises that are not yet implemented in Python track (the full list of them - https://github.com/exercism/python/issues/417#issuecomment-334841778).
You can help us change this situation by implementing this one. Ref #417

The guide ["Implement an exercise from specification"](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md) will help you succeed.

If you still have questions, I encourage you to ask them before starting working on the issue.

I also suggest claiming the issue by creating an empty PR as stated [here](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md#avoiding-duplicate-work)

Please note that there are some special ['closing issue' keywords](https://help.github.com/articles/closing-issues-using-keywords/) that you can put in description of your PR and Github will take care of closing the corresponding issue automatically after merge.
This simple thing helps us a lot, and makes our life a bit easier

### PR DESCRIPTION
**Pull request by PaulGlass**
resolves #729 
Want to implement this

### CODE REVIEW

**Comment by PaulGlass** [comment]
Done

File: config.json [hunk 1]
```diff
@@ -855,6 +855,14 @@
       ]
     },
     {
+      "uuid": "1818f134-0ed9-e680-9b29-45ffd2b3344b0f841c7",
+      "slug": "markdown",
+      "difficulty": 3,
+      "topics": [
+        "refactoring"
```
Comment (m-a-ge) [inline config.json]: Not sure that this topic is appropriate, could you please explain your reasoning?

You can use this list of general topics common among all tracks - https://github.com/exercism/problem-specifications/blob/master/TOPICS.txt

Comment (m-a-ge) [reply config.json]: `regular_expressions`, `pattern_matching` are good candidates for example

File: exercises/markdown/markdown.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import re
+
+
+def parse_markdown(markdown):
+    lines = markdown.split('\n')
```
Comment (m-a-ge) [inline exercises/markdown/markdown.py]: I think you forgot to replace an actual solution with an exercise placeholder for the learner.
Could you please fix this?

File: exercises/markdown/markdown_test.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import unittest
+from markdown import parse_markdown
+
+
```
Comment (m-a-ge) [inline exercises/markdown/markdown_test.py]: Could you please also leave a comment stating what version of `canonical-data.json` the tests were adopted as discussed in #784?

File: config.json (hunk 1 above)
Comment (PaulGlass) [reply config.json]: Look at [this file](https://github.com/exercism/problem-specifications/blob/master/exercises/markdown/description.md):
"The markdown exercise is a refactoring exercise. There is code that parses a given string with Markdown syntax and returns the associated HTML for that string. Even though this code is confusingly written and hard to follow, somehow it works and all the tests are passing! Your challenge is to re-write this code to make it easier to read and maintain while still making sure that all the tests keep passing."

File: exercises/markdown/markdown.py (hunk 2 above)
Comment (PaulGlass) [reply exercises/markdown/markdown.py]: Learner should refactor this code as I mentioned before.

File: config.json (hunk 1 above)
Comment (N-Parsons) [reply config.json]: @m-a-ge, this exercise is intended to be a refactoring exercise rather than an implementation one (see the README). This also explains the presence of a working solution (does TravisCI currently test this solution? I feel like it needs to) in `markdown.py`.

File: exercises/markdown/markdown_test.py (hunk 3 above)
Comment (N-Parsons) [reply exercises/markdown/markdown_test.py]: @PaulGlass, the new format is:
`# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0`

Comment (PaulGlass) [reply exercises/markdown/markdown_test.py]: Fixed

File: config.json (hunk 1 above)
Comment (m-a-ge) [reply config.json]: @PaulGlass sorry, I've missed it due to amount of opened PRs 😕 
@N-Parsons thanks!

No changes needed then 😄

File: exercises/markdown/markdown.py (hunk 2 above)
Comment (m-a-ge) [reply exercises/markdown/markdown.py]: @PaulGlass yeah, I see it now, thanks

**Comment by m-a-ge** [comment]
Thanks a lot for adding such an interesting exercise, @PaulGlass!
